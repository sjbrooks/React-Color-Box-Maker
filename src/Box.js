import React from "react";
import "./Box.css";

/** A single colored box used in the BoxList and NewBoxForm components.
 **/

function Box({ backgroundColor, width, height, id, remove }) {

  const handleRemove = () => remove(id);

  const styles = {
    backgroundColor,
    width: `${width}px`,
    height: `${height}px`
  }

  return (
  <div onClick={handleRemove} className="colored-box" id={id} style={styles}>
    <button>Remove</button>
  </div>
  )
}

export default Box;
