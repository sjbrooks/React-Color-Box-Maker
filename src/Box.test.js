import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box"

const testBox = < Box backgroundColor="cornflower" width="150" height="150"/>

test('renders Box without crashing ', () => {
    render(< Box />)
});

test("snapshot:  renders same testBox each time", function(){
    const{asFragment} = render(testBox);
    expect(asFragment()).toMatchSnapshot();
})

