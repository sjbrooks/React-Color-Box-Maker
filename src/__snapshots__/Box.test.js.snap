// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot:  renders same testBox each time 1`] = `
<DocumentFragment>
  <div
    class="colored-box"
    style="width: 150px; height: 150px;"
  >
    Remove
  </div>
</DocumentFragment>
`;
